<refentry id="libc---stack-chk-fail-1">

<refmeta>
<refentrytitle>__stack_chk_fail</refentrytitle>
<refmiscinfo>libc</refmiscinfo>
</refmeta>

<refnamediv>
<refname>__stack_chk_fail</refname>
<refpurpose>terminate a function in case of stack overflow</refpurpose>
<indexterm id="ix-libc---stack-chk-fail-1">
	<primary>__stack_chk_fail</primary></indexterm>
</refnamediv>
<refsynopsisdiv>
<funcsynopsis>
<funcsynopsisinfo>
#include &lt;libc.h&gt;
</funcsynopsisinfo><funcprototype>
<funcdef>void
<function>__stack_chk_fail</function>
</funcdef>
<void>
</funcprototype>
</funcsynopsis>
</refsynopsisdiv>
<refsect1>
<title>Description</title>
<para>
The interface <function>__stack_chk_fail</function>
shall abort the function that called it with a message that
a stack overflow has been detected.
The program that called the function shall then exit.
</para>
</refsect1>
<refsect1>
<title>Application Usage (informative)</title>
<para>
The interface <function>__stack_chk_fail</function>
does not check for a stack overflow itself.
It merely reports one when invoked.
</para>
</refsect1>
</refentry>
